src/AI/AIUtils.java

```diff
@@ -167,4 +168,24 @@ public static GameAction moveTowardLocation(Unit unit, XYCoord destination, Game
     }
     return stations;
   }
+  
+  /**
+   * Find a usable ability that has all specified flags, and add it to the provided queue.
+   */
+  public static CommanderAbility queueCromulentAbility(Queue<GameAction> q, Commander co, int flags)
+  {
+    CommanderAbility retVal = null;
+    ArrayList<CommanderAbility> abilities = co.getReadyAbilities();
+    for( CommanderAbility ab : abilities )
+    {
+      if( flags == (ab.AIflags & flags) )
```

Comment:
Might it make more sense to use OR instead of AND? If an ability is useful in more than one place, it will never be triggered here unless you pass in e.g. `CommanderAbility.PHASE_TURN_START | CommanderAbility.PHASE_TURN_END`, which seems silly on its face.
On the other hand, if these are meant to be mutually exclusive, then there's no reason to use flags.

Reply:
I'm pretty sure you're wrong.
The AND cuts out anything that's *not* in the flags you passed in. Then, if the ability satisfies all criteria, it gets used.
For instance, say I have a power that heals instantly, and another that heals after combat.
Ability 1 would be flagged as a healing ability and start/end.
Ability 2 would be flagged as a healing ability and start.
If I want a healing ability regardless of form, I can grab either.
If I'm at the start/end of my turn and want to filter for that, I can do that too.

Reply:
Hmm, yeah, I think you are correct. Carry on.